src/type/introspection.js

```diff
@@ -108,8 +109,7 @@ var __Type = new GraphQLObjectType({
         includeDeprecated: { type: GraphQLBoolean, defaultValue: false }
       },
       resolve(type, {includeDeprecated}) {
-        if (type instanceof GraphQLObjectType ||
-            type instanceof GraphQLInterfaceType) {
+        if (typeHasFields(type)) {
```

Comment:
This is not a safe change. Unions do not define any fields, I don't think it defines `getFields()`